### PR TITLE
chore: code clean up

### LINKED ISSUE #965: chore: include missing contract artifacts and rename test/precompiles to test/system-contract
### Problem

- Some contracts miss their artifacts
- test/precompiles needs to be renamed to test/system-contracts

### Solution

- Some contracts miss their artifacts
- test/precompiles needs to be renamed to test/system-contracts

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- Added missing artifacts.
- Renamed `test/precompiles` to `test/system-contract`.
- Replaced all instances of "precompile" with "system contract" (contract name, file names, etc.).
- Updated documentation for all system contracts.

**Related issue(s)**:

Fixes #965 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/system-contracts/exchange-rate/README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-## Exchange Rate Precompile methods
+## Exchange Rate System Contract Functions
 
-The following table highlights the coded Exchange rate precompile methods and their availability in the Hedera network exposed via the `0x168` address
+The Exchange Rate System Contract is accessible at address `0x168` on the Hedera network. This contract provides methods for retrieving and managing exchange rates between HBAR and fiat currencies.
```
Comment (lukelee-sl) [inline contracts/system-contracts/exchange-rate/README.md:3]: should fiat currencies -> US dollar?

File: contracts/system-contracts/hedera-account-service/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+## Hedera Account Service (HAS) System Contract Functions
+
+The Hedera Account Service (HAS) System Contract is accessible at address `0x16a` on the Hedera network. This contract introduces new account proxy contract for HBAR allowances, enables querying and granting approval of HBAR to a spender account from within smart contracts, allows developers to grant, retrieve, and manage HBAR allowances directly within smart contract code.
```
Comment (lukelee-sl) [inline contracts/system-contracts/hedera-account-service/README.md:3]: There is no description of what `isAuthorizedRaw` does?

Comment (lukelee-sl) [inline contracts/system-contracts/hedera-account-service/README.md:3]: Suggest something like - This contract introduces a new account proxy contract in order to interact with a contract for such functionality as HBAR allowances etc..

File: contracts/system-contracts/exchange-rate/README.md (hunk 1 above)
Comment (quiet-node) [reply contracts/system-contracts/exchange-rate/README.md]: Updated.

File: contracts/system-contracts/hedera-account-service/README.md (hunk 2 above)
Comment (quiet-node) [reply contracts/system-contracts/hedera-account-service/README.md]: Updated.

Comment (quiet-node) [reply contracts/system-contracts/hedera-account-service/README.md]: Updated.

File: contracts/system-contracts/hedera-token-service/README.md [hunk 3]
```diff
@@ -1,77 +1,79 @@
-## HTS Precompile methods
+## Hedera Token Service (HTS) System Contract Functions
 
-The following table highlights the coded HTS precompile methods and their availability in the Hedera network exposed via the `0x167` address.
+The Hedera Token Service (HTS) System Contract is accessible at address `0x167` on the Hedera network. This contract provides a comprehensive set of methods for creating, managing, and interacting with various types of tokens on the Hedera network.
```
Comment (Nana-EC) [inline contracts/system-contracts/hedera-token-service/README.md:3]: Not really a contract but an interface.
here and other locations
```suggestion
The Hedera Token Service (HTS) System Contract is accessible at address `0x167` on the Hedera network. This contract interface provides a comprehensive set of methods for creating, managing, and interacting with various types of tokens on the Hedera network.
```

**Review by lukelee-sl — APPROVED** [review]
LGTM

File: README.md [hunk 4]
```diff
@@ -8,31 +8,38 @@ The Hedera network utilizes system contracts at a reserved contract address on t
 These system contracts are precompiled smart contracts whose function selectors are mapped to defined network logic.
 In this way EVM users can utilize exposed HAPI features natively in their smart contracts.
 
-The precompiled functions are defined in this library and picked up by the [Hedera Services](https://github.com/hashgraph/hedera-services) repo.
+The system contract functions are defined in this library and picked up by the [Hedera Services](https://github.com/hashgraph/hedera-services) repo.
```
Comment (Nana-EC) [inline README.md:11]:
```suggestion
The system contract functions are defined in this library and implemented by the [Hedera Services](https://github.com/hashgraph/hedera-services) repo as part of consensus node functionality.
```

**Review by Nana-EC — COMMENTED** [review]
Nice changes

File: contracts/system-contracts/hedera-token-service/README.md (hunk 3 above)
Comment (quiet-node) [reply contracts/system-contracts/hedera-token-service/README.md]: Updated.

**Review by Nana-EC — APPROVED** [review]
🙌🏾

**Pull request merged**